src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -610,14 +610,19 @@ public override PSHostUserInterface UI
         /// <summary>
         /// See base class.
         /// </summary>
-        public void PushRunspace(Runspace newRunspace)
+        public void PushRunspace(Runspace runspace)
         {
             if (_runspaceRef == null)
             {
                 return;
             }
 
-            RemoteRunspace remoteRunspace = newRunspace as RemoteRunspace;
+            RemoteRunspace remoteRunspace = runspace as RemoteRunspace;
+            if (remoteRunspace is null)
+            {
+                throw new ArgumentException(ConsoleHostStrings.PushRunspaceNotRemote, nameof(runspace));
+            }
```

Comment:
Please use modern syntax:
```suggestion
            RemoteRunspace remoteRunspace = runspace as RemoteRunspace;
            if (remoteRunspace is RemoteRunspace remoteRunspace)
            {
                throw new ArgumentException(ConsoleHostStrings.PushRunspaceNotRemote, nameof(runspace));
            }
```

Reply:
I'm checking to see if it's not `RemoteRunspace`. I could do this instead but unsure if that's preferable or not.

```csharp
if (!(runspace is RemoteRunspace remoteRunspace))
```

Reply:
Ah! I updated my code.

Reply:
Nice, that's definitely clearer than before.